File: src/main/scala/booklet/impure/router/RootRouter.scala
package booklet.impure.router

import booklet.pure.http
import booklet.pure.views.RootView
import zhttp.http.Method.{GET, POST}
import zhttp.http._
import zio.json._

object RootRouter {
  val app: Http[Any, Nothing, Request, Response] = Http.collect[Request] {
    case GET -> !! =>
      http.CustomResponse.ok(body = Body.fromString(RootView.show.toString))
    case req @ POST -> !! / "api" / "authenticate" =>
      for {
        body <- req.bodyAsString
        data <- ZIO.fromEither(body.fromJson[Map[String, String]]).orElseFail(http.CustomResponse.badRequest("Invalid JSON"))
        idToken = data.getOrElse("id_token", "")
        userId = data.getOrElse("user_id", "")
        // Handle the ID token and user ID (e.g., store in session, validate, etc.)
      } yield http.CustomResponse.ok()
  }
}

File: src/main/ts/booklet/impure/router/RootRouter.ts
import { Http, Request, Response } from 'some-http-library';
import { CustomResponse } from '../../pure/http/CustomResponse';
import { RootView } from '../../pure/views/RootView';

export const RootRouter = {
  app: Http.collect<Request, Response>({
    'GET /': (req: Request) => {
      return CustomResponse.ok(RootView.show().toString());
    },
    'POST /api/authenticate': async (req: Request) => {
      try {
        const body = await req.bodyAsString();
        const data = JSON.parse(body);
        const idToken = data.id_token || '';
        const userId = data.user_id || '';
        // Handle the ID token and user ID (e.g., store in session, validate, etc.)
        return CustomResponse.ok();
      } catch (error) {
        return CustomResponse.badRequest('Invalid JSON');
      }
    }
  })
};
